### PR TITLE
[Bug] [command] Fix "Can only specify option -i once" in command line args parser

### LINKED ISSUE #1092: [Bug] [Core] fix "Can only specify option -i once" in command line args parser
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

`--variable` can only be specified once.
This bug was introduced in #813 when change command line parser.

### SeaTunnel Version

2.0.5

### SeaTunnel Config

```conf
env {
  spark.app.name = "SeaTunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
}

source {
  Fake {
    result_table_name = "my_dataset"
  }
}

transform {
   sql {
     sql = "select * from my_dataset where raw_message in ('"${name1}"', '"${name2}"')"
   }
}

sink {
  Console {}
}
```

### Running Command

```shell
./bin/start-seatunnel-spark.sh -c /tmp/spark.conf  -e client -m local -i name1="Hello" -i name2="World"
```

### Error Exception

```log
Exception in thread "main" com.beust.jcommander.ParameterException: Can only specify option -i once.
	at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:240)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:913)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:894)
	at com.beust.jcommander.JCommander.parseValues(JCommander.java:724)
	at com.beust.jcommander.JCommander.parse(JCommander.java:356)
	at com.beust.jcommander.JCommander.parse(JCommander.java:335)
	at org.apache.seatunnel.config.command.CommandLineUtils.parseSparkArgs(CommandLineUtils.java:29)
	at org.apache.seatunnel.SeatunnelSpark.main(SeatunnelSpark.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:849)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:167)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:195)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:924)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:933)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yx91490**
## Purpose of this pull request

close #1092
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by xleoken** [comment]
hi @yx91490, can you add a unit test? thanks.

**Comment by yx91490** [comment]
> hi @yx91490, can you add a unit test? thanks.

good suggestion, how do you think I should write? testing Jcommander's result seems meaningless.

**Comment by xleoken** [comment]
> > hi @yx91490, can you add a unit test? thanks.
> 
> good suggestion, how do you think I should write? testing Jcommander's result seems meaningless.

you can refer `org.apache.seatunnel.config.command.CommandLineUtilsTest`

**Comment by yx91490** [comment]
> hi @yx91490, can you add a unit test? thanks.

done

File: seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandFlinkArgs.java [hunk 1]
```diff
@@ -42,4 +44,7 @@ public boolean isTestConfig() {
         return testConfig;
     }
 
+    public List<String> getVariables() {
```
Comment (xleoken) [inline seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandFlinkArgs.java:47]: It seems that we never use `getVariables()` in main logic.

Comment (yx91490) [reply seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandFlinkArgs.java:47]: UT used.

Comment (yx91490) [reply seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandFlinkArgs.java:47]: > It seems that we never use `getVariables()` in main logic.

UT used.

File: seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandSparkArgs.java [hunk 2]
```diff
@@ -57,4 +59,11 @@ public boolean isTestConfig() {
         return testConfig;
     }
 
+    public String getMaster() {
```
Comment (asdf2014) [inline seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandSparkArgs.java:62]: Maybe we can assign a default value for the `master` variable to improve the ease of use, such as `local`. What do you think

Comment (yx91490) [reply seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/command/CommandSparkArgs.java:62]: make sense, but we can achive this goal in another issue.

**Comment by yx91490** [comment]
ping @leo65535 can be merged? @CalvinKirs

**Comment by xleoken** [comment]
also cc @garyelephant, seems we didn't use these variables in main logic.

**Comment by yx91490** [comment]
this pr was broken, continue to https://github.com/apache/incubator-seatunnel/pull/1236

**Pull request closed without merging**